Gems/AudioSystem/Code/Source/Engine/AudioSystem.cpp

```diff
@@ -85,86 +87,59 @@ namespace Audio
-        AudioSystemThreadSafeRequestBus::Handler::BusDisconnect();
-        AudioSystemInternalRequestBus::Handler::BusDisconnect();
+        #if !defined(AUDIO_RELEASE)
+        AzFramework::DebugDisplayEventBus::Handler::BusDisconnect();
+        #endif // !AUDIO_RELEASE
     }
 
     ///////////////////////////////////////////////////////////////////////////////////////////////////
-    void CAudioSystem::PushRequest(const SAudioRequest& audioRequestData)
+    void CAudioSystem::PushRequest(AudioRequestVariant&& request)
     {
-        CAudioRequestInternal request(audioRequestData);
-
-        AZ_Assert(g_mainThreadId == AZStd::this_thread::get_id(), "AudioSystem::PushRequest - called from non-Main thread!");
-        AZ_Assert(0 == (request.nFlags & eARF_THREAD_SAFE_PUSH), "AudioSystem::PushRequest - called with flag THREAD_SAFE_PUSH!");
-        AZ_Assert(0 == (request.nFlags & eARF_EXECUTE_BLOCKING), "AudioSystem::PushRequest - called with flag EXECUTE_BLOCKING!");
-
-        AudioSystemInternalRequestBus::QueueBroadcast(&AudioSystemInternalRequestBus::Events::ProcessRequestByPriority, request);
+        AZStd::scoped_lock lock(m_pendingRequestsMutex);
+        m_pendingRequestsQueue.push_back(AZStd::move(request));
     }
 
     ///////////////////////////////////////////////////////////////////////////////////////////////////
-    void CAudioSystem::PushRequestBlocking(const SAudioRequest& audioRequestData)
+    void CAudioSystem::PushRequests(AudioRequestsQueue& requests)
     {
-        // Main Thread!
-        AZ_PROFILE_FUNCTION(Audio);
-
-        CAudioRequestInternal request(audioRequestData);
-
-        AZ_Assert(g_mainThreadId == AZStd::this_thread::get_id(), "AudioSystem::PushRequestBlocking - called from non-Main thread!");
-        AZ_Assert(0 != (request.nFlags & eARF_EXECUTE_BLOCKING), "AudioSystem::PushRequestBlocking - called without EXECUTE_BLOCKING flag!");
-        AZ_Assert(0 == (request.nFlags & eARF_THREAD_SAFE_PUSH), "AudioSystem::PushRequestBlocking - called with THREAD_SAFE_PUSH flag!");
-
-        ProcessRequestBlocking(request);
-    }
-
-    ///////////////////////////////////////////////////////////////////////////////////////////////////
-    void CAudioSystem::PushRequestThreadSafe(const SAudioRequest& audioRequestData)
-    {
-        CAudioRequestInternal request(audioRequestData);
-
-        AZ_Assert(0 != (request.nFlags & eARF_THREAD_SAFE_PUSH), "AudioSystem::PushRequestThreadSafe - called without THREAD_SAFE_PUSH flag!");
-        AZ_Assert(0 == (request.nFlags & eARF_EXECUTE_BLOCKING), "AudioSystem::PushRequestThreadSafe - called with flag EXECUTE_BLOCKING!");
-
-        AudioSystemThreadSafeRequestBus::QueueFunction(AZStd::bind(&CAudioSystem::ProcessRequestThreadSafe, this, request));
+        AZStd::scoped_lock lock(m_pendingRequestsMutex);
+        for (auto& request : requests)
+        {
+            m_pendingRequestsQueue.push_back(AZStd::move(request));
+        }
```

Comment:
It would be move all the elements of the request container using a move iterator
```suggestion
        AZStd::scoped_lock lock(m_pendingRequestsMutex);
        for (auto& request : requests)
        {
            m_pendingRequestsQueue.insert(m_pendingRequestsQueue.end(), AZStd::make_move_iterator(request.begin()), AZStd::make_move_iterator(request.end()));
        }
```

Reply:
Doing it this way, I shouldn't need to use a loop, right?  i.e.

`m_pendingRequestsQueue.insert(m_pendingRequestsQueue.end(), AZStd::make_move_iterator(requests.begin()), AZStd::make_move_iterator(requests.end()));`

Reply:
Correct, the loop wouldn't be needed.

Reply:
I tried making this change, but ran into some compile issues.  I'll keep the original for now, and see if I can make progress on the compilation error.